diff --git a/rage/suite/src/cutscene/cutsmanager.h b/rage/suite/src/cutscene/cutsmanager.h
index 5e40bd98..ac0442b9 100644
--- a/rage/suite/src/cutscene/cutsmanager.h
+++ b/rage/suite/src/cutscene/cutsmanager.h
@@ -326,7 +326,7 @@ struct sEditCutfBlockingBoundsInfo
 //##############################################################################
 #define SETUP_CUTSCENE_WIDGET_FOR_CONTENT_CONTROLLED_BUILD  0
 #define SETUP_CUTSCENE_STREAMING_FOR_CONTENT_CONTROLLED_BUILD 0
-#define CUTSCENE_AUTHORIZED_FOR_PLAYBACK (!__FINAL)
+#define CUTSCENE_AUTHORIZED_FOR_PLAYBACK 0
 #define ALLOW_APPROVAL_RENDER (1) 
 
 
